test/metabase/models/dashboard_test.clj

```diff
@@ -634,6 +634,62 @@
      (testing "there are no \"Tab 3\""
        (is (false? (t2/exists? :model/DashboardTab :dashboard_id dashboard-id :name "Tab 3")))))))
 
+(deftest revert-dashboard-skip-archived-or-deleted-card-test
+  (t2.with-temp/with-temp [:model/Dashboard {dashboard-id :id}          {:name "A dashboard"}
+                           :model/Card      {will-be-archived-card :id} {}
+                           :model/Card      {will-be-deleted-card :id}  {}
+                           :model/Card      {unchanged-card       :id}  {}]
+    ;; 0. create a dashboard
+    (create-dashboard-revision! dashboard-id true)
+    ;; 1. add 3 cards and 1 text card
+    (t2/insert-returning-pks! :model/DashboardCard
```

Comment:
Using `insert-returning-pks!` adds extra unnecessary details to the test compared to `with-temp`. We don't need to use
the first revision at step 0. So instead of using `insert-returning-pks!` to create the dashboard cards you could just
create the dashcards using `with-temp`.

Reply:
I understand we can shorten this and have the dashboard card defined, but I want to simulate the test as close to how revision got built up as possible. If you see other tests in this namespace most of them do that. I think it's nice we keep it that way.

Reply:
I don't find it necessary and there are other tests that don't do it this way.
https://github.com/metabase/metabase/blob/f1bb2e70d2c2d3f933222a30267adfcf39555efe/test/metabase/models/dashboard_test.clj#L300

The main reason we should try to avoid using `t2/insert-returning-pks!` is because it doesn't use the temp defaults, so it introduces a dependency on the concrete representation of a dashcard, even when the details of the representation aren't relevant to the test.